### PR TITLE
Simple Payments: adds plan support check for Jetpack and WordPress.com.

### PR DESCRIPTION
**Pull request by rodrigoi**
This is blocking #9577

#### Changes proposed in this Pull Request:

Adds plan check for the Simple Payments shortcode. To prevent breaking the site if a user downgrades and has active Simple Payments buttons, the shortcode is ignored and removed from the post or page output.

#### Testing instructions:

* Apply this branch on a Jetpack site with a Premium or Business Subscriptions.
* Using Calypso, add a Simple Payment button to a page or post.
* Navigate to the page or post where the button was added.

The Simple Payment button should display without regressions on the page or post it was added to.

* Downgrade the site to the Free plan without deleting the SP button.
* Navigate to the Jetpack dashboard to refresh the site's settings.
* Navigate to the page or post where the SP button was added.

The page should not show a SP Button, nor it should display the unprocessed shortcode.

### CODE REVIEW

File: modules/simple-payments/simple-payments.php
Comment (kwight) [inline modules/simple-payments/simple-payments.php]: Can we call `remove_shortcode` instead here? It's always a little strange to have a `noop` method 🙃

File: modules/simple-payments/simple-payments.php
Comment (kwight) [inline modules/simple-payments/simple-payments.php]: Do we need to check this by blog sticker? It would be nicer if we check the plan on dotcom the same way we're checking the plan for JP. It would be _waay_ nicer if we could check both the same way but alas... 🙃

File: modules/simple-payments/simple-payments.php
Comment (kwight) [inline modules/simple-payments/simple-payments.php]: <3

File: class.jetpack.php
Comment (gwwar) [inline class.jetpack.php]: Is this the right spot for this check, since this feature is used for both wpcom and Jetpack? @dzver do you have any suggestions here? I believe we do have plan capabilities defined elsewhere.

File: modules/simple-payments/simple-payments.php
Comment (gwwar) [reply modules/simple-payments/simple-payments.php]: In wpcom we also have `is_blog_wpcom`, `is_blog_jetpack` available for use. Internally we can tell with a site_id value of 1 or 2

**Comment by gwwar** [comment]
👍 I verified manual steps here, eg simple payments display with premium plans and up. I also verified that on free/personal, that simple payments do not render, and that there are no errors displayed in logs or on screen.

screen shot 2018-05-30 at 2 26 16 pm

screen shot 2018-05-30 at 2 33 55 pm

Before landing, we'll want another 👍 from Jetpack folks, and see if this is the correct place to add a plan check.

File: class.jetpack.php
Comment (dzver) [reply class.jetpack.php]: It's a very common thing when implementing a new plan or a new plans-related endpoint to add a new way of defining features. We have many of these spread across the codebases. This $supports array is just one of many such implementations. It's hard to recommend one of the other implementations as superior. If you decide to refactor the whole thing, we can do a hangout with some other devs who worked on plans and agree on an official recommendation.

It doesn't make me happy to see this but the code is consistent with itself and I think is fine.

Comment (gwwar) [reply class.jetpack.php]: Thanks @dzver, 🤔we'll keep this for now, but I think it'll be worth revisiting later.

File: modules/simple-payments/simple-payments.php
Comment (rodrigoi) [reply modules/simple-payments/simple-payments.php]: I've been looking for a better solution, but using `has_blog_sticker` is a stablished pattern for checking plans in WP.com (premium themes, AT eligibility, all use stickers), and is also following the same patterns already in use to check for WP.com and plan support on Jetpack.
This is mostly because we are doing this with a _Jetpack First_ approach.

File: modules/simple-payments/simple-payments.php
Comment (rodrigoi) [reply modules/simple-payments/simple-payments.php]: The problem is that `remove_shortcode` just removes the hooks associated with the shortcode, but it will still show the shortcode as content.

| `remove_shortcode` | `add_shortcode(..., noop)`|
| - | - |
| screen shot 2018-05-31 at 00 12 27 | screen shot 2018-05-31 at 00 13 12 |

My intention is not to break the site showing a shortcode as text.

File: class.jetpack.php
Comment (rodrigoi) [reply class.jetpack.php]: This was not my first option when implementing this, but since `simple-payments` is not a full Jetpack module (it's loaded by `modules-extras.php`) we can't use the automatic mapping of plan features to modules (using the `/sites/{site_id}/features` endpoint), or the automatic deactivation check on `check_plan_deactivate_modules`.  I had to treat it as an exception.

If we refactor SP into a complete Jetpack module, we can remove it from the exception list. But I think that exceeds the scope of this PR

I have a diff (D14013-code) that adds SP as a plan feature.

**Comment by Copons** [comment]
This LGTM! ✨

**Review by gwwar — APPROVED** [review]

File: modules/simple-payments/simple-payments.php
Comment (oskosk) [inline modules/simple-payments/simple-payments.php]: This should be updated to read 6.3.0 now

File: modules/simple-payments/simple-payments.php
Comment (mdawaffe) [inline modules/simple-payments/simple-payments.php]: If it's the blog owner (maybe just any admin?) viewing the page, it'd be nice if this could output something like "Simple Payments is not supported by your Jetpack Plan. To learn more, and to upgrade to a supported plan, visit _these resources_. (This message is only shown to site administrators.)"

**Comment by oskosk** [comment]
I pushed a commit updating the version from `6.2.0` to `6.3.0` in the docs for the `jetpack_disable_simple_payments` filter.

**Review by oskosk — APPROVED** [review]
This one tested well for me. I relabeled as needs review because the bot removed the label. This could be fine to go after addressing MIke's feedback. IMO it could be addressed in another PR too if this one PR is blocking another. If going this way, @rodrigoi can you add an issue with the proposal that Mike mantioned?

**Comment by rodrigoi** [comment]
@oskosk Thanks for the review! I've added issue #9777 to our backlog to add WP-Admin or Admin user only warnings when for some reason the user still has published products after a downgrade.

File: modules/simple-payments/simple-payments.php
Comment (Copons) [reply modules/simple-payments/simple-payments.php]: +1 on this!

For example, this is the message I used for the Twitter Timeline widget deprecation:
https://github.com/Automattic/jetpack/blob/758a77efd4c64e9469b2e544cb4ce1c7d7b5ef95/modules/widgets/twitter-timeline.php#L92-L93

**Review by Copons — COMMENTED** [review]
This LGTM, but I'd really love if we replaced the SP buttons with a message visible to admins explaining why they disappeared.
All in all, we can't assume that folks owning an upgrade know exactly what's included in that plan.

**Pull request merged**
